TeaLeaf/src/com/tealeaf/TeaLeafApplication.java

```diff
@@ -16,12 +16,24 @@
 import android.app.Application;
 import android.content.Context;
 import com.tealeaf.plugin.PluginManager;
+import android.content.ComponentCallbacks2;
 
-public class TeaLeafApplication extends  Application {
+public class TeaLeafApplication extends Application implements ComponentCallbacks2 {
 	@Override
 	public void onCreate() {
 		PluginManager.init(this);
 		PluginManager.callAll("onCreateApplication", this.getApplicationContext());
 	}
 
+	@Override
+	public void onLowMemory() {
+		NativeShim.textureManagerMemoryWarning();
+	}
+
+	@Override
+	public void onTrimMemory(int level) {
+		if (level == TRIM_MEMORY_RUNNING_CRITICAL) {
```

Comment:
IMO, Any memory warning should be taken seriously and resources released, not only when it is critical